cpp/src/parquet/encoder.cc

```diff
@@ -170,6 +179,7 @@ template <typename DType>
 void PlainEncoder<DType>::Put(const T* buffer, int num_values) {
   if (num_values > 0) {
     PARQUET_THROW_NOT_OK(sink_.Append(buffer, num_values * sizeof(T)));
+    unencoded_data_bytes_ += num_values * sizeof(T);
```

Comment:
this is only needed for consistency?

Reply:
Yes, I think it is pretty straight-forward to implement other types. I don't want to add a byte array specific function or variable to the generic `Encoder` and `EncoderImpl` classes, which is ugly.